data/json/ui/widgets-structured/problem-entries.json

```diff
@@ -0,0 +1,24 @@
+[
+  {
+    "id": "structured_weariness_transition_graph",
```

Comment:
When moving the entries of theis file to the main sidebar file, I get

```
ERROR : src/generic_factory.h:446 [const T& generic_factory<T>::obj(const string_id<T>&) const [with T = widget]] invalid widgets id "structured_weariness_transition_graph"
```

Not sure what is wrong here. What stands out is that they all use `copy-from`, and that the base entries all use the `colors` property.

Reply:
Due to game limitations, the sidebar files are loaded in alphabetical order. You cannot copy from a widget lower down on the list.

Reply:
So this refers to the `copy-from`, not the actual usage of a widget I guess? So the problem is that `sidebar-structured.json` comes before `speed.json`, `stamina.json` and `weariness.json`?

So possible solutions would be to

* put the widgets into the named files; probably not a good idea as they are quite specific (like setting a certain width)
* use a sub-directory (which are loaded later) for stuff specific to a certain sidebar; this is what I started the branch from, and given the relative complexity of this sidebar I found it more structured this way; however, this would break the current pattern of all sidebar-specific stuff in the bar's main file

Reply:
You could also rename the sidebars JSON file to start with Z. That would also work

Reply:
> Due to game limitations, the sidebar files are loaded in alphabetical order.

I think `deferred_json` can be used to solve this limitation,

Reply:
> I think `deferred_json` can be used to solve this limitation,

@Qrox Not sure what you mean here. Does this refer to changing how ui JSON is loaded, in the C++ code?

Reply:
Yes, item definitions are loaded using `DynamicDataLoader::deferred_json` so I think it can be used for sidebar preset too.

Reply:
Ok, thanks! Will have a look into it.

Reply:
Turns out that my current understanding of the codebase is not sufficient to solve this. Thus, opened an issue for it: #59202.

Reply:
As a workaround, modified the affected widgets to not use `copy-from`.